pkg/amqpcommand/command_client.go

```diff
@@ -76,8 +82,9 @@ func (c *CommandClient) Start() {
 				return
 			default:
 				err := c.doProcess()
+				atomic.StoreInt32(&c.connected, 0)
 				if err != nil {
-					atomic.StoreInt32(&c.connected, 0)
+					c.lastError = err
```

Comment:
Doesn't this need to be a atomic.StorePointer()?

Reply:
Its only used by the same thread AFAICT, whereas connected is used by both 'client request' thread, and the 'processor' thread.

Reply:
So it is.  Comment withdrawn.